px-breadcrumbs.es6.js

```diff
@@ -72,224 +64,137 @@
-          let strArrayShortened = breadcrumbsObj.shortenedItems;
-          
-          this._set_mainPathItems(strArrayShortened);
-          return;
-        }
+      /*
+      * Option 1:
+      * Check to see if the container (which is sized automatically to fill out the page)
+      * can fit all the items in the breadcrumbs.
+      */
+      if (ulWidth > breadcrumbsObj.sizeOfFullBreadcrumbs) {
+        this._set_mainPathItems(itemPath);
+        return;
+      }
+
+      /*
+      * Option 2:
+      * Find out if the container can now fit all the
+      * shortened items, plus the last item that wasn't shortened.
+      */
+      if (ulWidth > breadcrumbsObj.sizeOfAllShortenedItemsExcludingLastItem + breadcrumbsObj.sizeOfFullLastItem) {
+        let strArrayShortenedWithFullLastItem = breadcrumbsObj.allShortenedItemsExcludingLast.concat(breadcrumbsObj.lastItemFull);
+        this._set_mainPathItems(strArrayShortenedWithFullLastItem);
+        return;
+      }
 
-        /*
-        * option 4
-        * we have to create an array with overflow.
-        * we only get to this if non of the if statements above are true.
-        */
-      
-        this._set_mainPathItems(this._createArrayWithOverflow(itemPath, _ulWidth, breadcrumbsObj));
+      /*
+      * Option 3:
+      * Check if can fit after shortening all the items.
+      */
+      if (ulWidth > breadcrumbsObj.sizeOfAllShortenedItems) {
+        let strArrayShortened = breadcrumbsObj.shortenedItems;
+        this._set_mainPathItems(strArrayShortened);
+        return;
+      }
+
+      /*
+      * Option 4:
+      * Not all of the shortened options will fit. Have to create an array with overflow.
+      */
+      this._set_mainPathItems(this._createArrayWithOverflow(itemPath, ulWidth, breadcrumbsObj));
 
     },
+
     /*
-    * this method is called once we've established that we need to have an
-    * array with overflow.
-    * we keep removing the size of items - starting from the beginning of the array - 
-    * from the total size of all the items, until we can fit everything + the last item that isn't shortened
-    * into the container.
-    * @param {Array} strArray the array that holds the breadcrumbs
-    * @param {number} _ulWidth the width of the ul container
+    * Called once it's established that we need to have an array with overflow.
+    * Keep removing the size of each item - starting from the beginning of the array -
+    * from the total size of all the items, until we can fit everything, plus the last item that
+    * isn't shortened into the container.
     */
     _createArrayWithOverflow(strArray, _ulWidth, breadcrumbsObj) {
-      
+
       var pointer = 0,
           currentAccumSize = breadcrumbsObj.sizeOfAllShortenedItemsExcludingLastItem,
           sizeOfFullLastItem = breadcrumbsObj.sizeOfFullLastItem,
-          sizeOfEllipsis = breadcrumbsObj.sizeOfEllipsis,
+          sizeOfEllipsis = 16,
```

Comment:
The overflow object uses the "..." icon now:

![screen shot 2017-07-28 at 1 00 48 pm](https://user-images.githubusercontent.com/2130613/28734245-d2e1da34-7394-11e7-9b3b-6619ee762eea.png)

Although I just noticed that sizeOfEllipsis is 16 in one place and 36 in another, that's problematic...